### PR TITLE
#161255374 Highlight and Comment on any text in an article

### PR DESCRIPTION
**Pull request by njeri-ngigi**
#### What does this PR do?
1. Adds CRUD functions for highlighting and commenting on any text in an article:
- User can highlight text
- Users can add comments with the highlight
- Users can retrieve highlighted text
- Users can retrieve all highlights made on an article
- Users can update their highlights and comments
- Users can delete their highlights and comments
##### Endpoints involved:
 ```
/api/articles/slug/highlights/
/api/articles/slug/
```
2. Adds tests for all the CRUD methods
#### How to manually test these features
- Clone the Repo
- Make Migrations
- python manage.py migrate
- Use the endpoints to test on postman
#### Description of tasks to be completed
As a user i should be able to highlight any text in an article(created by me or not) and comment on it.

#### What are the relevant pivotal tracker stories?
[#161255374 Users should be able to highlight and comment on any text in an article](https://www.pivotaltracker.com/story/show/161255374)

#### Screenshots
image
image

### CODE REVIEW

**Review by AnguleMathias — APPROVED** [review]
great work!

**Review by maxwellgithinji — APPROVED** [review]
LGTM

**Review by Jwesh — APPROVED** [review]
line length should not be more than 80 characters.. check on that..

File: authors/apps/articles/tests/test_highlight.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import json
+from rest_framework.views import status
+from .base import BaseTest
+
+class HighlightTestCase(BaseTest):
+    '''class representing tests for highlighting and commenting on text in an article'''
+    def test_highlight(self):
+        '''test creating highlights and comments'''
+        # test successful highlight and comment
+        response = self.client.post(self.HIGHLIGHT, self.highlight_data, HTTP_AUTHORIZATION=self.non_user_token, format="json")
```
Comment (3N61N33R) [inline authors/apps/articles/tests/test_highlight.py]: Check on pep-8 linting for this line and the whole file. Some lines seem to have exceeded the acceptable limit

**Review by 3N61N33R — CHANGES_REQUESTED** [review]

File: authors/apps/articles/tests/test_highlight.py (hunk 1 above)
Comment (njeri-ngigi) [reply authors/apps/articles/tests/test_highlight.py]: Review the changes @graycadeau

**Review by 3N61N33R — APPROVED** [review]
great, lgtm

**Pull request merged**